src/utils/base-url-from-env-extractor.ts

```diff
@@ -0,0 +1,89 @@
+/*
+ * SPDX-FileCopyrightText: 2022 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+import { Optional } from '@mrdrogdrog/optional'
+import type { BaseUrls } from '../components/common/base-url/base-url-context-provider'
+import { Logger } from './logger'
+import { isTestMode } from './test-modes'
+
+/**
+ * Extracts the editor and renderer base urls from the environment variables.
+ */
+export class BaseUrlFromEnvExtractor {
+  private baseUrls: Optional<BaseUrls> | undefined
+  private logger = new Logger('Base URL Configuration')
+
+  private extractUrlFromEnvVar(envVarName: string, envVarValue: string | undefined): Optional<URL> {
+    return Optional.ofNullable(envVarValue)
+      .filter((value) => {
+        const endsWithSlash = value.endsWith('/')
+        if (!endsWithSlash) {
+          this.logger.error(`${envVarName} must end with an '/'`)
```

Comment:
Why not simply add a trailing slash at this point?

Reply:
I don't wanna "guess" the URL. The admin should set the correct one.

Reply:
I don't think this has anything to do with guessing, it's simply adding the required slash.

Reply:
It's not "simple"

`https://example.org/hdiwig` isn't a directory URL. It's a file. I have no idea if the admin wants to put the instance under `https://example.org/` or if he is missing a slash.

So..  no. I won't guess 😅

Reply:
Ok